In version 3.3.0 upstream started making use of features introduced
in macOS 10.11, which caused failures when building the MacOSX
backend on earlier macOS versions..

see: https://github.com/matplotlib/matplotlib/pull/17956

and the discussion about why upstream's solution does not work for us;
the current patch is as Ryan suggested here::
https://github.com/macports/macports-ports/commit/b07e0eb6440b1499bfd2157e19db86fc79984471#comments

--- src/_macosx.m.orig	2021-03-31 03:54:14.000000000 -0400
+++ src/_macosx.m	2021-04-03 12:45:49.000000000 -0400
@@ -1210,10 +1210,12 @@
     rect.size.height = 0;
     rect.origin.x += height;
     NSTextView* messagebox = [[NSTextView alloc] initWithFrame: rect];
-    if (@available(macOS 10.11, *)) {
+
+    #if MAC_OS_X_VERSION_MAX_ALLOWED >= 101100 && MAC_OS_X_VERSION_MIN_REQUIRED >= 101100
         messagebox.textContainer.maximumNumberOfLines = 2;
         messagebox.textContainer.lineBreakMode = NSLineBreakByTruncatingTail;
-    }
+    #endif
+
     [messagebox setFont: font];
     [messagebox setDrawsBackground: NO];
     [messagebox setSelectable: NO];
